### ISSUE TITLE
Interactive rebase in git exists after first chunk when neovim is set as the default editor

### ISSUE THREAD

**Issue by deivid-rodriguez** [neovim/neovim#2705]
If I set neovim as the default git editor, and I choose edit in a rebase chunk, all the rest of the chunks are skipped after existing nvim. So,

Steps:

* `git config --global core.editor nvim`
* `git add -p`
* Type `e` to open nvim and edit a chunk.
* Edit the diff.
* Type `x` to save the new diff.

Expected output:
* I get a prompt for the next chunk.

Actual output:
* The rebase tool exits.

I assume this is a bug in `neovim` because it works using just `vim`. I also think it's a recent regression because it didn't use to happen until very recently.

```
NVIM 0.0.0-alpha (compiled May 19 2015 15:20:27)
Commit: ?
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/build/config -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/src -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/usr/include/luajit-2.0 -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/.deps/usr/include -I/usr/include -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/build/src/nvim/auto -I/build/buildd/neovim-0.0.0+git201505190944+1638+9+10~ubuntu14.04.1/build/include
Compiled by buildd@lcy01-14

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/nvimrc"
     user vimrc file: "~/.nvimrc"
 2nd user vimrc file: "~/.nvim/nvimrc"
      user exrc file: "~/.exrc"
  fall-back for $VIM: "/usr/share/nvim"
```

Thanks!

**Comment by oni-link** [comment]
Perhaps related to #2598? Does this PR fix your problem?

**Comment by fwalch** [comment]
Might be the same as #2377.

**Comment by deivid-rodriguez** [comment]
Yes, it does! Thanks!!

**Comment by justinmk** [comment]
I'll finish that PR tonight.

**Comment by fwalch** [comment]
@justinmk Awesome!

**Comment by deivid-rodriguez** [comment]
:+1:

**Comment by fwalch** [comment]
Should have been fixed by neovim/neovim#2598, please report back if not.

**Comment by edi9999** [comment]
Had that same problem with git add --patch, but it is fixed by that pull request . Thanks !
